erpnext/accounts/doctype/payment_entry/payment_entry.json

```diff
@@ -282,6 +282,34 @@
   {
    "allow_on_submit": 0, 
    "bold": 0, 
+   "collapsible": 0, 
+   "columns": 0, 
+   "depends_on": "eval:doc.mode_of_payment==\"Check\"", 
+   "description": "If name different from Party name", 
+   "fieldname": "name_on_check", 
```

Comment:
Instead of adding a new field, allow editing party_name after submit and remove the field from hidden.